assumes/expression.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package assumes
+
+import "github.com/juju/version/v2"
+
+var (
+	_ Expression = (*FeatureExpression)(nil)
+	_ Expression = (*CompositeExpression)(nil)
+)
```

Comment:
I never get why these aren't in test files? As they're asserts on the implementations, you don't need them to be checked every time you run.

Reply:
These ones are compile-time assertions; they don't trigger allocations.

Reply:
Also you don't want them in tests because you could still build a binary without running the tests